benchmarks/planetarium_benchmarks.cpp

```diff
@@ -0,0 +1,245 @@
+    goes_8_elements.inclination = 0.4232 * Degree;
+    goes_8_elements.longitude_of_ascending_node = 97.7420 * Degree;
+    goes_8_elements.eccentricity = 0.0004776;
+    goes_8_elements.argument_of_periapsis = 192.8349 * Degree;
+    goes_8_elements.mean_anomaly = 121.5613 * Degree;
+    goes_8_elements.mean_motion = 1.00264613 * (2 * π * Radian / Day);
+
+    ephemeris_->Prolong(goes_8_epoch);
+    KeplerOrbit<Barycentric> const goes_8_orbit(
+        *earth_, MasslessBody{}, goes_8_elements, goes_8_epoch);
+    goes_8_trajectory_.Append(
+        goes_8_epoch,
+        ephemeris_->trajectory(earth_)->EvaluateDegreesOfFreedom(goes_8_epoch) +
+            goes_8_orbit.StateVectors(goes_8_epoch));
+    auto goes_8_instance = ephemeris_->NewInstance(
+        {&goes_8_trajectory_},
+        Ephemeris<Barycentric>::NoIntrinsicAccelerations,
+        HistoryParameters());
+    CHECK_OK(ephemeris_->FlowWithFixedStep(goes_8_epoch + 100 * Day,
+                                           *goes_8_instance));
+  }
+
+  DiscreteTrajectory<Barycentric> const& GOES8Trajectory() const {
+    return goes_8_trajectory_;
+  }
+
+  Planetarium MakePlanetarium(
+      Perspective<Navigation, Camera> const& perspective) const {
+    // No dark area, human visual acuity, wide field of view.
+    Planetarium::Parameters parameters(
+        /*sphere_radius_multiplier=*/1,
+        /*angular_resolution=*/0.4 * ArcMinute,
+        /*field_of_view=*/90 * Degree);
+    return Planetarium(parameters,
+                       perspective,
+                       ephemeris_.get(),
+                       earth_centred_inertial_.get());
+  }
+
+ private:
+  Ephemeris<Barycentric>::FixedStepParameters EphemerisParameters() {
+    return Ephemeris<Barycentric>::FixedStepParameters(
+        SymmetricLinearMultistepIntegrator<QuinlanTremaine1990Order12,
+                                           Position<Barycentric>>(),
+        /*step=*/10 * Minute);
+  }
+
+  Ephemeris<Barycentric>::FixedStepParameters HistoryParameters() {
+    return Ephemeris<Barycentric>::FixedStepParameters(
+        SymmetricLinearMultistepIntegrator<Quinlan1999Order8A,
+                                           Position<Barycentric>>(),
+        /*step=*/10 * Second);
+  }
+
+  not_null<std::unique_ptr<SolarSystem<Barycentric>>> const solar_system_;
+  not_null<std::unique_ptr<Ephemeris<Barycentric>>> const ephemeris_;
+  not_null<MassiveBody const*> const earth_;
+  not_null<std::unique_ptr<NavigationFrame>> const earth_centred_inertial_;
+  DiscreteTrajectory<Barycentric> goes_8_trajectory_;
+};
+
+}  // namespace
+
+void RunBenchmark(benchmark::State& state,
+                  Perspective<Navigation, Camera> const& perspective) {
+  Satellites satellites;
+  Planetarium planetarium = satellites.MakePlanetarium(perspective);
+  RP2Lines<Length, Camera> lines;
+  int total_lines = 0;
+  int iterations = 0;
+  constexpr Instant now = "2000-01-21T04:41:30,5"_TT;
```

Comment:
Where does this date come from?

Reply:
It's an eclipse that happened to be in January of 2000.